@model System.Web.Mvc.HandleErrorInfo

<hgroup class="title">
    <h1 class="error">Oooops!!</h1>
</hgroup>


<h2 class="error">Something didn't work properly</h2>

<p>
    Please click <span id="exitLink" ></span> and try the operation again.
@*    @if (User.Identity.IsAuthenticated && User.IsInRole("user"))
    {
        @Html.ActionLink("here to return to your list of Surveys", "List", "Survey")
    }
    else
    {
        @Html.ActionLink("here to return to the Home page", "Logoff", "Account")
    }*@
    
</p>
<p>
    Click 
    <a id="showException" href="#" >here, to see the details of the problem</a>.
</p>

<div id="exceptionInfo" >
    <div>
        @Html.DisplayNameFor(m => m.ControllerName)
        @Html.DisplayFor(m => m.ControllerName)
    </div>
    <div>
        @Html.DisplayNameFor(m => m.ActionName)
        @Html.DisplayFor(m => m.ActionName)
    </div>

    <div>
        @Html.DisplayNameFor(m => m.Exception)
        @Html.DisplayFor(m => m.Exception)
    </div>
</div>

<script>
    $(document).ready(function () {
        //  Set up event handlers.
        $("#showException").click(function () {
            $("#exceptionInfo").show();
        });

        //  Setup the link to exit this error page.
        //  We're trying to avoid the WebSecurity related InvalidOperationExceptions 
        //  because the localDB has timed out (I think this is the reason).
        if ($('#logoutForm').length > 0) {
            $('#exitLink').html('<a href="javascript:document.getElementById(\'logoutForm\').submit()">here to return to the Home page</a>');
        } else {
            $('#exitLink').html('@Html.ActionLink("here to return to the Home page", "Index", "Home")');
        }

        //  Hide the exception stuff.
        $("#exceptionInfo").hide();
    });
</script>


